firestore/angular-rewrite/functions/src/index.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    onDocumentWritten,
+} from "firebase-functions/v2/firestore";
+import * as logger from "firebase-functions/logger";
+import { getFirestore } from "firebase-admin/firestore";
+import { initializeApp } from "firebase-admin/app";
+
+import { Restaurant } from '../../types/restaurant';
+import { Rating } from '../../types/ratings';
+
+initializeApp();
+const db = getFirestore();
+
+export const updateNumRatings = onDocumentWritten(
```

Comment:
Let's remove this from the current PR and add it in an upcoming PR. There's some changes I think could improve this a bit:

Remove use of global variables by returning values in the promise and chaining the promises.
Remove the use of finally if we can get away with it

Comment liberally - for example it's unclear the difference between numRatingsReported and actualRatings. IIUC numRatingsReported are pre-aggregated but it's unclear in what cases we expect them to be different.

Bonus: for good form we should make the code robust to race conditions and describe via comments how we accomplish this

Reply:
Sure thing; I'll make these changes and move all functions files to a separate PR!